test/unittests/libprecompiled/test_WorkingSealerManagerPrecompiled.cpp

```diff
@@ -0,0 +1,148 @@
+            fixture->context, bytesConstRef(&in), sealerAccount, sealerAccount));
+        fixture->getSealerList();
+        // check the result is equal
+        BOOST_CHECK(workingSealers == fixture->m_workingSealerList);
+        BOOST_CHECK(pendingSealers == fixture->m_pendingSealerList);
+        BOOST_CHECK(fixture->m_pendingSealerList.size() == 0);
+    }
+
+    // case2: valid proof, but the origin is not exist in the workingSealer
+    LOG(INFO) << LOG_DESC(
+        "testRotateWorkingSealerWithoutRotate: valid proof, but the origin is not exist in the "
+        "workingSealer");
+    auto nonSealerAccount = dev::toAddress(dev::KeyPair::create().pub());
+    BOOST_CHECK_THROW(fixture->workingSealerManagerPrecompiled->call(
+                          fixture->context, bytesConstRef(&in), nonSealerAccount, nonSealerAccount),
+        PrecompiledException);
+
+    // case3: invalid input(must be lastest hash)
+    LOG(INFO) << LOG_DESC("testRotateWorkingSealerWithoutRotate: valid proof, invalid input");
+    in = abi.abiIn(WSM_METHOD_ROTATE_STR, fixture->vrfPublicKey, toHex(dev::sha3("test")),
+        vrfInputWithProof.second);
+    BOOST_CHECK_THROW(fixture->workingSealerManagerPrecompiled->call(
+                          fixture->context, bytesConstRef(&in), nonSealerAccount, nonSealerAccount),
+        PrecompiledException);
+
+    // case4: invalid public key(the origin is not one of the sealers)
+    LOG(INFO) << LOG_DESC("testRotateWorkingSealerWithoutRotate: invalid public key");
+    in = abi.abiIn(WSM_METHOD_ROTATE_STR, toHex(bytesConstRef{fixture->m_keyPair.pub().data(), 64}),
+        vrfInputWithProof.first, vrfInputWithProof.second);
+    BOOST_CHECK_THROW(fixture->workingSealerManagerPrecompiled->call(
+                          fixture->context, bytesConstRef(&in), nonSealerAccount, nonSealerAccount),
+        PrecompiledException);
+
+    // case5: invalid proof
+    LOG(INFO) << LOG_DESC("testRotateWorkingSealerWithoutRotate: invalid proof");
+    auto fakedKeyPair = dev::KeyPair::create();
+    auto fakedVRFKeyPair = fixture->generateVRFKeyPair(fakedKeyPair);
+    auto fakedVRFInputWithProof =
+        fixture->generateVRFProof(fixture->context, fakedVRFKeyPair.first);
+    in = abi.abiIn(WSM_METHOD_ROTATE_STR, fixture->vrfPublicKey, fakedVRFInputWithProof.first,
+        fakedVRFInputWithProof.second);
+    BOOST_CHECK_THROW(fixture->workingSealerManagerPrecompiled->call(
+                          fixture->context, bytesConstRef(&in), sealerAccount, sealerAccount),
+        PrecompiledException);
+}
+
+BOOST_AUTO_TEST_CASE(testWithRotate)
+{
+    auto fixture = std::make_shared<WorkingSealerManagerFixture>();
+    fixture->initWorkingSealerManagerFixture(10, 4, 1);
+    fixture->getSealerList();
+    auto orgWorkingSealers = fixture->m_workingSealerList;
+    auto orgPendingSealers = fixture->m_pendingSealerList;
+    auto orgSealerList = fixture->sealerList;
```

Comment:
Codacy found an issue: [Variable 'orgSealerList' is assigned a value that is never used.](https://app.codacy.com/gh/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=5737534)